yt/visualization/plot_window.py

```diff
@@ -868,8 +868,12 @@ def _setup_plots(self):
             axis_index = self.data_source.axis
 
             xc, yc = self._setup_origin()
-            if self.ds.unit_system.name.startswith("us"):
-                # this should happen only if the dataset was initialized with argument unit_system="code"
+            if self.ds.unit_system.name.startswith("us") or self.ds.no_cgs_equiv_length:
+                # this should happen only if the dataset was initialized with
+                # argument unit_system="code" or if it's set to have no CGS
+                # equivalent.  This only needs to happen here in the specific
+                # case that we're doing a computationally intense operation
+                # like using cartopy, but it prevents crashes in that case.
                 (unit_x, unit_y) = ('code_length', 'code_length')
```

Comment:
I think I wrote the original version of this `if` to be able to display "code length" as a unit in x and y labels for 2D plots. Unless I'm missing something here, the change reads to me like this it's now always go to be what's in those axis labels when using certain geometries, including `"spherical"`, and I feel like it's out of scope here.

(In truth I'm almost certain I'm missing something)

Reply:
Well, so, hm.

It turns out that `unit_x` and `unit_y` get used for setting the values.  The ultimate result of this is that at some point, instead of using the lat/lon, they get converted to some other intermediate unit, and this messes stuff up.

*But*, you should be able to manually change them, at a later time, I believe.

Reply:
As discussed on Slack, I just checked that my suspicions were unfounded. You leave me no choice but to approve the change :-)